src/lib/configs.ts

```diff
@@ -102,4 +102,31 @@ export function getConfigs(): { [key: string]: string } {
     GLEE_CONFIG_FILE_PATH,
     ASYNCAPI_FILE_PATH
   }
-}
\ No newline at end of file
+}
+
+
+export class GlobalGleeConfig {
+  private _config: GleeConfig
+
+  constructor(config: GleeConfig) {
+    this._config = config
+  }
+
+  async init(protocol: string){
+
+    const defaultValues = {
+      docs: {
+        enabled: true,
+        folder: 'docs',
+        template: '@asyncapi/markdown-template'
+      }
+    }
+
+    return {
+      ...this._config[protocol],
+      defaultValues[protocol]
+    }
+  }
```

Comment:
I don't think we can do this, as this is currently giving me errors and like this we have to traverse the whole object to find what values to return according to the default or passed, this creates the same problem as it tries to parse the entire object, this has the same issue as https://github.com/asyncapi/glee/issues/436